linkcheck/models.py

```diff
@@ -327,16 +327,20 @@ def check_external(self, external_recheck_interval=EXTERNAL_RECHECK_INTERVAL):
             'timeout': LINKCHECK_CONNECTION_ATTEMPT_TIMEOUT,
         }
         try:
-            if self.url.count('#'):
-                # We have to get the content so we can check the anchors
+            # At first try a HEAD request
+            response = requests.head(self.external_url, **request_params)
```

Comment:
Do you think it's fine to always do a HEAD request first? This will prevent big binary files from being downloaded, but it also causes one additional HEAD request for all HTML files with hash anchors (which will probably be more frequent than binary files with hash anchors)...

Reply:
Alright, I opened #154 for the first commit.